lib/async/redis/sentinels.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module Async
+	module Redis
+		class SentinelsClient < Client
```

Comment:
While I'm fine with this name for now, please give me a day to think about it.

There are two models for naming things like this:

- `Client::Sentinels`

or as you did

- `SentinelsClient`

The decision is based on how many other kinds of "clients" we expect to have and their semantics.

Reply:
Not sure what other clients we might want to have, maybe one for Redis Cluster.

I don't have a strong preference about this I'm fine with whatever you decide.

This is kind of related with something that I wanted to discuss but forgot to mention in the description of the PR. Do you think we should instantiate this client with `Client::Sentinels.new(...)` / `SentinelsClient.new` or instead, do you think it'd be better to instantiate it with `Client.new(...)`? With the second option, the current Client initializer would be responsible for checking the options received in the params, and if there are some sentinels there, instantiate internally a SentinelsClient and return it instead of the standard one.

Reply:
What do you think makes the most sense in the context of the user?

I always try to consider user/interface ergonomics... but I also try to avoid making things inefficient/add layers of abstraction where it *only* serves ergonomics purposes.

If users would *always* know they have sentinels or not, it makes sense to have separate classes, for example.

We shouldn't do anything to make `Client` "inefficient". It's the fast path, everything needs to be explicit.

With `Async::HTTP` I realised it's cumbersome for many use cases, so I introduced `Async::HTTP::Internet` which solves those challenges for the user. Some kind of multiplexing layer which understands there might be sentinels, replicas, and other things. Under the hood you still end up going via a directly connected client, but the abstraction is resolved one layer above.

Reply:
The other benefit to that style of implementation, is that if you want to change your policy, it's not inheritance but composition. I really prefer composition if possible, simple layers bring about complex behaviour rather than complex class hierarchies that are hard to reason about.

Reply:
> If users would _always_ know they have sentinels or not, it makes sense to have separate classes, for example.

They always know, it cannot be transparent because the user needs to provide the list of sentinels.

Reply:
@ioquatix I agree with the point you made about composition. The idea would be to instantiate an `Async::Redis::Client` in the new class and then delegate everything to that instance, right?
I think we can't do that without changes in the `Async::Redis::Client` class. We'd need a way to be able to set the block that we use in the `connect` method.